overlord/state/taskrunner.go

```diff
@@ -28,6 +28,18 @@ import (
 	"github.com/snapcore/snapd/logger"
 )
 
+// HookFunc is the type of function that can be registered as a hook.
+type HookFunc func(chg *Change)
```

Comment:
there could be also Task hooks in theory?  maybe they should be called ChangeHook* here and the other places

Reply:
Makes sense, updated